NServiceBus.FileBasedRouting.Tests/EndpointBasedIntegrationTests.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using NServiceBus.Configuration.AdvanceExtensibility;
+using NServiceBus.FileBasedRouting.Tests.Contracts.Commands;
+using NServiceBus.Routing;
+using NUnit.Framework;
+
+namespace NServiceBus.FileBasedRouting.Tests
+{
+    public class EndpointBasedIntegrationTests
+    {
+        const string XmlA = @"
+ <endpoints>
+  <endpoint name=""EndpointName"">
+    <handles>
+        <command type = ""NServiceBus.FileBasedRouting.Tests.Contracts.Commands.A, NServiceBus.FileBasedRouting.Tests.Contracts"" />
+    </handles>
+  </endpoint>
+ </endpoints>
+ ";
+        const string XmlB = @"
+ <endpoints>
+  <endpoint name=""EndpointName"">
+    <handles>
+        <command type = ""NServiceBus.FileBasedRouting.Tests.Contracts.Commands.B, NServiceBus.FileBasedRouting.Tests.Contracts"" />
+    </handles>
+  </endpoint>
+ </endpoints>
+ ";
+        const string EndpointName = "EndpointName";
+
+        [Test]
+        public async Task Should_read_default_file()
+        {
+            using (Use("endpoints.xml", XmlA))
+            {
+                var routing = await GetRouting(null);
+
+                var routeA = routing(typeof(A));
+                Assert.AreEqual(EndpointName, routeA.Endpoint);
+
+                var routeB = routing(typeof(B));
+                Assert.IsNull(routeB);
+            }
+        }
+
+        [Test]
+        public async Task Should_read_custom_file()
+        {
+            const string name = "custom.xml";
+            using (Use(name, XmlB))
+            {
+                var routing = await GetRouting(name);
+
+                var routeA = routing(typeof(A));
+                Assert.IsNull(routeA);
+                
+                var routeB = routing(typeof(B));
+                Assert.AreEqual(EndpointName, routeB.Endpoint);
+            }
+        }
+
+        static async Task<Func<Type,UnicastRoute>> GetRouting(string filePath)
+        {
+            var endpointConfiguration = new EndpointConfiguration("test");
+            endpointConfiguration.UsePersistence<InMemoryPersistence>();
+            endpointConfiguration.SendFailedMessagesTo("error");
+            endpointConfiguration.UseTransport<MsmqTransport>();
+
+            if (filePath == null)
+            {
+                endpointConfiguration.EnableFileBasedRouting();
+            }
+            else
+            {
+                endpointConfiguration.EnableFileBasedRouting(filePath);
+            }
+
+            await Endpoint.Create(endpointConfiguration); // to init all features
+            var routingTable = endpointConfiguration.GetSettings().Get<UnicastRoutingTable>();
```

Comment:
let's keep them in here as long as they don't cause any pain since they have been written now anyway. But good to 🔥 them immediately on some sign of issues